system/Database/BaseConnection.php

```diff
@@ -588,15 +599,17 @@ public function addTableAlias(string $table)
 	 * Should automatically handle different connections for read/write
 	 * queries if needed.
 	 *
-	 * @param string  $sql
-	 * @param mixed   ...$binds
-	 * @param boolean $setEscapeFlags
-	 * @param string  $queryClass
+	 * @param mixed       $sql
+	 * @param mixed       ...$binds
+	 * @param boolean     $setEscapeFlags
+	 * @param string|null $queryClass
 	 *
 	 * @return BaseResult|Query|false
 	 */
-	public function query(string $sql, $binds = null, bool $setEscapeFlags = true, string $queryClass = 'CodeIgniter\\Database\\Query')
+	public function query($sql, $binds = null, bool $setEscapeFlags = true, ?string $queryClass = null)
 	{
+		is_null($queryClass) && $queryClass = $this->queryClass;
```

Comment:
Aside from the possible BC break in the method signature, this line can use the Elvis operator.
```suggestion
		$queryClass = $queryClass ?: $this->queryClass;
```